tests/users/tests_get_user_details_api.py

```diff
@@ -0,0 +1,147 @@
+from django.http import response
+from users.models import CustomUser
+from rest_framework.test import APITestCase, APIClient
+from rest_framework import status
+
+class GetUserDetailsTestCase(APITestCase):
+
+    get_user_details_api = "/users/get_user_details/"
+    
+    def setUp(self):
+        self.email = "crash.test.dummy@gmail.com"
+        self.password = "test.modelx"
+
+        self.user = CustomUser.objects.create_user(
+            email = self.email,
+            username = self.email,
+            first_name = "Crash",
+            last_name = "Test",
+            contact = "+919999999999",
+            password = self.password,
+        )
+
+    def create_auth_client(self):
+        login_api = "/users/login/"
+        login_payload = {
+            'email': self.email,
+            'password': self.password
+        }
+
+        auth_client = APIClient()
+        login_response = auth_client.post(login_api, login_payload)
+        auth_token = login_response.data['token']
+        auth_client.credentials(HTTP_AUTHORIZATION=auth_token)
+        return auth_client
+
+    def test_success_authorized_user(self):
+        auth_client = self.create_auth_client()
+        response = auth_client.get(self.get_user_details_api)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```

Comment:
We can also test whether the data returned matches the user's details. Adding something like:
```
        data_user = {
            'first_name': 'Crash',
            'last_name': 'Test',
            'email': self.email,
            'verified': False,
            'contact': '+919999999999',
            'bquiz_score': 0,
            'user_type': 'GST',
            'linkedin': None,
            'facebook': None,
            'applied': False,
            'id': 1
        }
        self.assertEqual(response.data, data_user)
```